zipkin/src/tracer.cc

```diff
@@ -111,4 +112,51 @@ void Tracer::reportSpan(Span &&span) {
 void Tracer::setReporter(ReporterPtr reporter) {
   reporter_ = std::move(reporter);
 }
+
+uint64_t Tracer::getDroppedSpanCount() const {
```

Comment:
If we want to expose these as part of the `Reporter`, we wouldn't have to cast anything.

Reply:
Yes, that would be better.

Even if you want to cast, `reinterpret_cast` isn't the [right one](https://github.com/isocpp/CppCoreGuidelines/blob/master/CppCoreGuidelines.md#Rh-dynamic_cast) to use.